marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/EntitySearchControllerTest.java

```diff
@@ -57,30 +71,198 @@ void testCRUDOnSavedQuery() throws Exception {
+            .andDo(result -> {
+                ArrayNode response = readJsonArray(result.getResponse().getContentAsString());
+                assertTrue(response.size() > 0, "There should be at least one saved query document");
+            });
 
         // testing delete savedQuery document
-        deleteJson(SAVED_QUERIES_PATH+"/query", params)
-                .andExpect(status().isNoContent());
-
-        getJson(SAVED_QUERIES_PATH+"/query", params)
-                .andExpect(status().isOk())
-                .andDo(result -> {
-                    ObjectNode response = readJsonObject(result.getResponse().getContentAsString());
-                    assertTrue(response.isEmpty());
-                });
+        deleteJson(SAVED_QUERIES_PATH + "/query", params)
+            .andExpect(status().isNoContent());
+
+        getJson(SAVED_QUERIES_PATH + "/query", params)
+            .andExpect(status().isOk())
+            .andDo(result -> {
+                ObjectNode response = readJsonObject(result.getResponse().getContentAsString());
+                assertTrue(response.isEmpty());
+            });
+    }
+
+
+    @Test
+    void testRowExport() throws Exception {
+        installReferenceProject(true);
+
+        Customer customer1 = new Customer(1, "Jane", 123456789, "2012-05-16");
+        Customer customer2 = new Customer(2, "John", 987654321, "2013-06-16");
+        createCustomer(customer1);
+        createCustomer(customer2);
+
+        waitForTasksToFinish();
+
+        loginAsUser(TestConstants.OPERATOR_USERNAME);
+
+        String json = "{\n" +
+            "    \"savedQuery\": {\n" +
+            "        \"id\": \"\",\n" +
+            "        \"name\": \"some-query\",\n" +
+            "        \"description\": \"some-query-description\",\n" +
+            "        \"query\": {\n" +
+            "            \"searchText\": \"\",\n" +
+            "            \"entityTypeIds\": [\n" +
+            "                \"Customer\"\n" +
+            "            ],\n" +
+            "            \"selectedFacets\": {\n" +
+            "                \"Collection\": {\n" +
+            "                    \"dataType\": \"xs:string\",\n" +
+            "                    \"stringValues\": [\n" +
+            "                        \"Customer\"\n" +
+            "                    ]\n" +
+            "                }\n" +
+            "            }\n" +
+            "        },\n" +
+            "        \"propertiesToDisplay\": [\n" +
+            "            \"customerId\",\n" +
+            "            \"name\",\n" +
+            "            \"customerNumber\"\n" +
+            "        ]\n" +
+            "    }\n" +
+            "}";
+
+        int limit = 2;
+        int totalColumns = 3;
+
+        // Export using query document
+        postWithParams(EXPORT_PATH, getRequestParams(limit, json))
+            .andExpect(request().asyncStarted())
+            .andDo(MvcResult::getAsyncResult)
```

Comment:
Nice - is this verifying that the async support is configured correctly then?

Reply:
Yes, `request().asyncStarted()` checks for the async support. 
`getAsyncResult()` is needed to wait for the async result to return.